pkg/api/types.go

```diff
@@ -715,6 +717,31 @@ func (p *Properties) HasVirtualMachineScaleSets() bool {
 	return false
 }
 
+// HasZonesForAllAgentPools returns true if all of the agent pools have zones
+func (p *Properties) HasZonesForAllAgentPools() bool {
```

Comment:
nit: this function could break as soon as it finds an agent pool without AZ similar to what  https://github.com/Azure/acs-engine/pull/3864/files#diff-0226d5bd49c7195fda1784e5dd4274c9R573 does
```
for _, agentPoolProfile := range p.AgentPoolProfiles { 
			if !agentPoolProfile.HasAvailabilityZones() { 
				return false
			} 
return true
```